src/aerialmap_display.cpp

```diff
@@ -242,35 +244,41 @@ void AerialMapDisplay::createGeometry()
 
 void AerialMapDisplay::update(float, float)
 {
-  // create all geometry, if necessary
+  // update tiles, if necessary
   assembleScene();
-
-  // draw
-  context_->queueRender();
+  // transform map-fixed scene object into fixed frame
+  transformMapTileToFixedFrame();
 }
 
 void AerialMapDisplay::navFixCallback(sensor_msgs::NavSatFixConstPtr const& msg)
 {
-  ref_fix_ = *msg;
+  // check if update is necessary
+  TileId newCenterTileID{ tile_url_, fromWGSCoordinate({ msg->latitude, msg->longitude }, zoom_), zoom_ };
+  bool const centerTileChanged = (!lastCenterTile_ || !(newCenterTileID == *lastCenterTile_));
 
-  // re-load imagery
-  received_msg_ = true;
-  loadImagery();
-  transformAerialMap();
+  if (centerTileChanged)
+  {
+    updateCenterTile(newCenterTileID, msg);
+  }
 }
 
-void AerialMapDisplay::loadImagery()
+void AerialMapDisplay::updateCenterTile(TileId const& newCenterTileID, sensor_msgs::NavSatFixConstPtr const& msg)
 {
   if (!isEnabled())
   {
     return;
   }
 
-  // When the plugin starts, the properties from the config are set.
```

Comment:
the whole startup and callback logic changed quite a bit. I can not easily translate this comment onto the new structure. Also
"// When the plugin starts, the properties from the config are set.  // This will call the callbacks and these will call this function." .. is how the RViz works. One usually knows that :) If not, I'd have to put that into every property change cb